cmake/StaticBuild.cmake

```diff
@@ -58,6 +77,10 @@ if(BUILD_STATIC)
       set(GNUTLS_LIBRARIES "${GNUTLS_LIBRARIES} -lcrypt32")
       # And sockets
       set(GNUTLS_LIBRARIES "${GNUTLS_LIBRARIES} -lws2_32")
+      # link ncrypt and unistring statically
+      set(GNUTLS_LIBRARIES "${GNUTLS_LIBRARIES} -Wl,-Bstatic -lncrypt -Wl,-Bdynamic")
+      # p11-kit only available as dynamic library for MSYS2 on Windows and dynamic linking of unistring is required
+      set(GNUTLS_LIBRARIES "${GNUTLS_LIBRARIES} -lp11-kit -lunistring")
```

Comment:
These three aren't on all systems so they still need to be conditional.

Reply:
I made p11-kit and unistring conditional.